### PR TITLE
Move to API v9 and add threads support

### PR DESCRIPTION
**Pull request by greenbigfrog**
This PR scopes moving to API v9, with the main focus on having thread support.
(Some breaking changes such as `/guilds/embed` moving to `/guilds/widget` is tracked in #25 )

### CODE REVIEW

**Comment by greenbigfrog** [comment]
This should be ready for review now.

**Comment by PixeLInc** [comment]
Okay, pushed some fixes and whatnot.

I think this should be good to go now.
I am leaving it until later tonight (~15-20 hours) for anyone to comment or point out anything wrong, or suggest something that could be better.

Otherwise, we mergin'

File: src/discordcr/mappings/gateway.cr [hunk 1]
```diff
@@ -84,6 +84,11 @@ module Discord
       DirectMessages         = 1 << 12
       DirectMessageReactions = 1 << 13
       DirectMessageTyping    = 1 << 14
+
+      # Generates an Unprivileged intents constant, removing GuildMembers and GuildPresences.
+      {% begin %}
+        Unprivileged = {{ @type.constants.reject { |e| ["All", "None", "GuildMembers", "GuildPresences"].includes?(e.stringify) }.join("|").id }}
```
Comment (Daniel-Worrall) [inline src/discordcr/mappings/gateway.cr:90]:
```suggestion
        alias Unprivileged = Union({{ *@type.constants.reject { |e| ["All", "None", "GuildMembers", "GuildPresences"].includes?(e.stringify) } }})
```

**Comment by PixeLInc** [comment]
Thanks for the PR!

A follow-up PR should probably be created, to deal with your aforementioned embed->widget thing along with a PR to add any other missing v8/v9 features that are not covered here.

And if anyone finds any issues with this that I did not find from my testing, please do open an issue or scold me within the discordcr dapi channel.

**Pull request merged**